Modelica/Electrical/Analog/Semiconductors.mo

```diff
@@ -671,10 +699,13 @@ Stefan Vorkoetter - new model proposed.</li>
     Capcje = smooth(1, Cje*powlin(veb/Phie, Me));
     ccb = smooth(1, Taur*is_t/(NR*vt_t)*exlin2(vcb/(NR*vt_t), EMin, EMax) + Capcjc);
     ceb = smooth(1, Tauf*is_t/(NF*vt_t)*exlin2(veb/(NF*vt_t), EMin, EMax) + Capcje);
-    C.i = icb/br_t + ccb*der(vcb) + Ccs*der(C.v) + (icb - ieb)*qbk;
+    C.i = icb/br_t + ccb*der(vcb) + (icb - ieb)*qbk - iS;
     B.i = -ieb/bf_t - icb/br_t - ceb*der(veb) - ccb*der(vcb);
-    E.i = -B.i - C.i + Ccs*der(C.v);
-
+    E.i = -B.i - C.i - iS;
+    iS = -Ccs * (der(C.v) - der(vS));
```

Comment:
Same comment as for npn, maybe iS = -Ccs * der(vcs) would be more clear.

Reply:
Yes indeed, this makes sense